handle-no-inverse-relationship

Mechabear opened the pull request:
- Adds a condition to check if the inverse key exists before proxying the relationships;

Commit:
handle-no-inverse-relationship

- Adds a condition to check if the inverse key exists before proxying the relationships;

Commit:
handle-no-reverse-relationship

- Separates applyChanges from the inverseKey condition;

Shadowere reviewed (changes requested):
Add test cases for models with no inverse keys

Shadowere reviewed (changes requested):
la section dans `_setupRelationship` qui get la inverseKey pourrait être moved dans la classe de model-extractor (c'est pas là que les propriétés des models sont extracted).
Une méthode extract/getInverseKey(modelName, relationshipName) pourrait être nice et sa serait plus clean

Shadowere commented on addon/services/model-proxy.js:
```diff
@@ -125,7 +131,12 @@ export default Service.extend({
   _setupRelationship(name, descriptor, modelDefinition, proxy, model, proxyRelationships) {
     let store = get(this, 'store');
-    let inverseKey = modelDefinition.inverseFor(name, store).name;
+    let inverseConfig = modelDefinition.inverseFor(name, store);
```
Here (pour le extract/getInverseKey)

Shadowere commented on addon/utils/model-proxy.js:
```diff
@@ -130,7 +130,12 @@ export default EmberObject.extend(Evented, {
       modelDefinition.eachRelationship((name, descriptor) => {
         if (!ignoredRelationships.includes(name)) {
-          let inverseKey = modelDefinition.inverseFor(name, store).name;
+          let inverseConfig = modelDefinition.inverseFor(name, store);
```
And here (pour le extract/getInverseKey)

Commit:
handle-no-inverse-relationship

- Adds a getInverseKey function in the model extractor;
- Uses the getInverseKey function where it is needed;
- Adds a no-inverse-model for tests;
- Adds a test to make sure it does not crash when applying changes on a proxy with no inverse relationship;

Shadowere commented on addon/services/model-extractor.js:
```diff
@@ -60,5 +60,9 @@ export default Service.extend({
+  },
+
+  getInverseKey(modelDefinition, relationshipName) {
+    return modelDefinition.inverseFor(relationshipName, this.get('store'));
```
Tu peux faire toute la logique pour `get` le name de ton inverseKey ou return null directement ici

Commit:
handle-no-inverse-relationship

- Adds more content to getInverseKey function;
